src/stores/wpcom-api.ts

```diff
@@ -0,0 +1,70 @@
+import { createApi } from '@reduxjs/toolkit/query/react';
+import { z } from 'zod';
+import { useOffline } from 'src/hooks/use-offline';
+import type { BaseQueryFn, FetchBaseQueryError } from '@reduxjs/toolkit/query';
+import type WPCOM from 'wpcom';
+
+const welcomeMessageSchema = z.object( {
+	messages: z.array( z.string() ),
+	example_prompts: z.array( z.string() ),
+} );
+
+export type WelcomeMessageResponse = z.infer< typeof welcomeMessageSchema >;
+
+let wpcomClient: WPCOM | undefined;
+
+export const setWpcomClient = ( client: WPCOM | undefined ) => {
+	wpcomClient = client;
+};
+
+const wpcomBaseQuery: BaseQueryFn<
+	{ path: string; apiNamespace?: string },
+	unknown,
+	FetchBaseQueryError
+> = async ( args ) => {
+	try {
+		const response = await wpcomClient!.req.get( args );
+		return { data: response };
+	} catch ( error ) {
+		return {
+			error: {
+				status: 500,
+				data: error,
+			},
+		};
+	}
+};
+
+export const wpcomApi = createApi( {
+	reducerPath: 'wpcomApi',
+	baseQuery: wpcomBaseQuery,
+	tagTypes: [ 'WelcomeMessages' ],
+	endpoints: ( builder ) => ( {
+		getWelcomeMessages: builder.query< WelcomeMessageResponse, void >( {
+			query: () => ( {
+				path: '/studio-app/ai-assistant/welcome',
+				apiNamespace: 'wpcom/v2',
+			} ),
+			transformResponse: ( response: unknown ) => {
+				return welcomeMessageSchema.parse( response );
+			},
+			keepUnusedDataFor: 60 * 60,
+		} ),
+	} ),
+} );
+
+function withConnectionChecks< TArg, TResult >(
+	useQueryHook: ( arg: TArg, options?: { skip?: boolean } ) => TResult
+): ( arg: TArg, options?: { skip?: boolean } ) => TResult {
```

Comment:
```suggestion
function withConnectionChecks< TResult, TArg >(
	useQueryHook: TypedUseQuery< TResult, TArg, typeof wpcomBaseQuery >
): TypedUseQuery< TResult, TArg, typeof wpcomBaseQuery > {
```
Took me a frustratingly long time to think of the right type definition here, but I think this would be a more complete definition.

Reply:
Thank you for improving it, it is indeed a more complete definition! 👍